python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1677,27 +1677,25 @@ def _should_backfill_atomic_asset_subset_unit(
             )
         ).compute_intersection(entity_subset_to_filter)
 
-        not_waiting_for_parent_subset = entity_subset_to_filter.compute_difference(
-            possibly_waiting_for_parent_subset
-        )
```

Comment:
before, we filtered out possibly_waiting_for_parent_subset and then potentially added a subset of it back in. Now that it is all or nothing, we can just selectively remove it based on the result of _get_cant_run_with_parent_reason i think, and not filter anything out if there is no failure reason